.github/workflows/run_docker_container.yml

```diff
@@ -0,0 +1,84 @@
+name: Pull and Run Docker Container
+
+on:
+  workflow_dispatch:
+    inputs:
+      number-jobs:
+        description: "Number of concurrent jobs"
+        required: true
+        default: "1"
+  push:
+
+
+env:
+  number-jobs: 1
+  lichess_bot_token: ${{ secrets.LICHESS_BOT_TOKEN }}
+
+concurrency:
+  group: run
+  cancel-in-progress: true
+
+jobs:
+  prejob:
+    runs-on: self-hosted
+    outputs:
+      job_matrix: ${{ steps.set-matrix.outputs.job_matrix }}          
+    steps:
+      - name: Use python3 to convert number-jobs to a list
+        id: set-matrix
+        run: |
+          # send telegram message that we are starting a job, send the json config
+          matrix_value=$(python3 -c "print(str(list(range(0, ${{ github.event.inputs.number-jobs || env.number-jobs }} ))).replace(' ', ''))")          
+          echo "job_matrix=$matrix_value" >> $GITHUB_OUTPUT
+
+      - name: Check if folder exists
+        run: test -d "${{ github.workspace }}/adam-chess-data"
+        id: check-folder
+        continue-on-error: true
+
+      - name: Setup Git LFS
+        run: |
+          # if file exists do NOT download:
+          if [ $? -eq 0 ]; then
+            echo "No need to clone"
+          else
+            curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash
+            sudo apt-get install git-lfs
+            git lfs install
+            git lfs clone https://huggingface.co/datasets/ezipe/adam-chess-data
+          fi  
+
+  run:
+    timeout-minutes: 6 # 24 hours
+    runs-on: self-hosted
+    container:
+      image: vincentlinzhu11/chess_transformer_mothership:latest
+      volumes:
+        - ${{ github.workspace }}/adam-chess-data/original:/chess-nanoGPT/data/lichess_hf_dataset  # apply a volume mount
```

Comment:
this shouldn't map directly into lichess_hf_dataset - it should go into `/dataset`, and then you should symbolic link to the files inside `/dataset`